src/game/server/entities/character.cpp

```diff
@@ -1432,7 +1432,7 @@ void CCharacter::HandleSkippableTiles(int Index)
 			{
 				if(Direction.x > 0.0000001f)
 					SpeederAngle = -std::atan(Direction.y / Direction.x);
-				else if(Direction.x < 0.0000001f)
+				else if(Direction.x < -0.0000001f || (GameServer()->EmulateBug(BUG_SPEEDTILE_MAXSPEED) && Direction.x < 0.0000001f))
```

Comment:
This also changes the behavior when `BUG_SPEEDTILE_MAXSPEED` is set. The first term of the logical or will be evaluated regardless of `BUG_SPEEDTILE_MAXSPEED` being set.

```suggestion
				else if(Direction.x < (GameServer()->EmulateBug(BUG_SPEEDTILE_MAXSPEED) ? 0.0000001f : -0.0000001f))
```

Same below.

Reply:
As we discussed in discord, our versions are logically the same, but yours appears to be a bit cleaner (in terms of readability).